lib/rucio/daemons/conveyor/throttler.py

```diff
@@ -149,86 +152,111 @@ def __schedule_requests():
+                    result_dict[dest_rse_id]['activities'][activity]['waiting'] += counter
+                    result_dict[dest_rse_id]['waiting'] += counter
                 else:
-                    result_dict[activity][dest_rse_id]['accounts'][account]['transfer'] += counter
-                    result_dict[activity][dest_rse_id]['transfer'] += counter
-
-        for activity in result_dict:
-            for dest_rse_id in result_dict[activity]:
-                rse_name = result_dict[activity][dest_rse_id]['rse']
-                availability = get_rse(rse_name).availability
-                if availability & 2:  # dest_rse is not blacklisted for write
-                    threshold = result_dict[activity][dest_rse_id]['threshold']
-                    transfer = result_dict[activity][dest_rse_id]['transfer']
-                    waiting = result_dict[activity][dest_rse_id]['waiting']
-                    if waiting:
-                        logging.debug("Request status for %s at %s: %s" % (activity, rse_name,
-                                                                           result_dict[activity][dest_rse_id]))
-                    if threshold is None:
-                        logging.debug("Throttler remove limits(threshold: %s) and release all waiting requests for activity %s, rse_id %s" % (threshold, activity, dest_rse_id))
-                        delete_rse_transfer_limits(rse=None, activity=activity, rse_id=dest_rse_id)
-                        release_waiting_requests(rse=None, activity=activity, rse_id=dest_rse_id)
-                        record_counter('daemons.conveyor.throttler.delete_rse_transfer_limits.%s.%s' % (activity, rse_name))
-                    elif transfer + waiting > threshold:
-                        logging.debug("Throttler set limits for activity %s, rse %s" % (activity, rse_name))
-                        set_rse_transfer_limits(rse=None, activity=activity, rse_id=dest_rse_id, max_transfers=threshold, transfers=transfer, waitings=waiting)
-                        record_gauge('daemons.conveyor.throttler.set_rse_transfer_limits.%s.%s.max_transfers' % (activity, rse_name), threshold)
-                        record_gauge('daemons.conveyor.throttler.set_rse_transfer_limits.%s.%s.transfers' % (activity, rse_name), transfer)
-                        record_gauge('daemons.conveyor.throttler.set_rse_transfer_limits.%s.%s.waitings' % (activity, rse_name), waiting)
+                    result_dict[dest_rse_id]['activities'][activity]['accounts'][account]['transfer'] += counter
+                    result_dict[dest_rse_id]['activities'][activity]['transfer'] += counter
+                    result_dict[dest_rse_id]['transfer'] += counter
+
+        for dest_rse_id in result_dict:
+            dest_rse_release_strategy = config_core.get('throttler_release_strategy', 'dest_%s' % dest_rse_id, default='grouped_fifo', use_cache=False)
```

Comment:
Maybe the default should be fifo?

Reply:
yes, makes sense. Than its easier to update to the next release.